src/connectors/refinement-list/__tests__/connectRefinementList-test.js

```diff
@@ -457,4 +458,305 @@ describe('connectRefinementList', () => {
 
     expect(rendering.lastCall.args[0].hasExhaustiveItems).toEqual(false);
   });
+
+  it('can search in facet values', () => {
+    const widget = makeWidget({
+      attributeName: 'category',
+      limit: 2,
+    });
+
+    const helper = algoliasearchHelper(
+      fakeClient,
+      '',
+      widget.getConfiguration({})
+    );
+    helper.search = sinon.stub();
+    helper.searchForFacetValues = sinon.stub().returns(
+      Promise.resolve({
+        exhaustiveFacetsCount: true,
+        facetHits: [
+          {
+            count: 33,
+            highlighted: 'Salvador <em>Da</em>li',
+            value: 'Salvador Dali',
+          },
+          {
+            count: 9,
+            highlighted: '<em>Da</em>vidoff',
+            value: 'Davidoff',
+          },
+        ],
+        processingTimeMS: 1,
+      })
+    );
+
+    // Sinulate the lifecycle
+    widget.init({
+      helper,
+      state: helper.state,
+      createURL: () => '#',
+      onHistoryChange: () => {},
+    });
+    expect(rendering.callCount).toBe(1);
+
+    widget.render({
+      results: new SearchResults(helper.state, [
+        {
+          hits: [],
+          facets: {
+            category: {
+              c1: 880,
+            },
+          },
+        },
+        {
+          facets: {
+            category: {
+              c1: 880,
+            },
+          },
+        },
+      ]),
+      state: helper.state,
+      helper,
+      createURL: () => '#',
+    });
+    expect(rendering.callCount).toBe(2);
+    // Simulation end
+
+    const search = rendering.lastCall.args[0].searchForItems;
+    search('da');
+
+    const [
+      sffvFacet,
+      sffvQuery,
+      maxNbItems,
+      paramOverride,
+    ] = helper.searchForFacetValues.lastCall.args;
+
+    expect(sffvQuery).toBe('da');
+    expect(sffvFacet).toBe('category');
+    expect(maxNbItems).toBe(2);
+    expect(paramOverride).toEqual({
+      highlightPreTag: undefined,
+      highlightPostTag: undefined,
+    });
+
+    // because of how promises are resolved we use setImmediate
+    setImmediate(() => {
```

Comment:
You should probably use `done` for finish the test. I think the assertions inside `setImmediate` are executed after the test, if they fail all the process will fail instead of just the test case.

You can ensure how many assertions you expect in an async test with:

```js
expect.assertions(numberOfAssertionsExpected);
```

Reply:
I didn't have any issue when the tests are green, it's just that when they fail it breaks jest completely. I will try that, maybe it's gonna be fine.

Reply:
I wonder also if we could now use:

```js
await Promise.resolve();
await Promise.resolve();
```

As an easy way to wait for the render fn to be called

Reply:
You are right @vvo we could do use an `async` function to `await` the result of the `search` function.

Reply:
But you can still use await Promise.resolve() to force the event loop to run the render function. That's what setImmediate does here.

Was just a small comment if it's feasible it might be easier to read than setImmediate (and safer)

Reply:
That's what I call an interesting hack @vvo :) Thanks.

Reply:
After looking at some jest issues, it seems that it handles returned promises and since my main issue with setImmediate is that when the test fail jest breaks, I ended up creating a new promise and in the then do my tests. It's not as smart but `¯\_(ツ)_/¯`